seonshine_frontend/src/pages/menuManagement/components/ModalMenuItem.tsx

```diff
@@ -0,0 +1,331 @@
+
+import { avatarBaseURL } from '@/constants/image';
+import { CreateMenuItemPayloadType, GetMenuListResponseType, UpdateMenuItemPayloadType } from '@/types/user';
+import { isValidImageFile } from '@/utils/file';
+
+import { useCreateMenuItemApi, useDeleteMenuItemApi, useUpdateMenuItemApi } from '@/apis/hooks/userApi.hook';
+
+import { menuListInfoSchema, MenuListInfoSchemaType } from '../schema';
+import { approvalItemDelete } from './constant';
+
+type ModalMenuItemPropsType = {
+  isOpen: boolean;
+  selectedItem?: GetMenuListResponseType;
+  setSelectedItem: (selectedItem: GetMenuListResponseType | null) => void;
+  setIsModalOpen: (isOpen: boolean) => void;
+  isCreateModal?: boolean;
+  item_id?: string;
+  restaurant_id?: string;
+};
+const ModalMenuItem = ({
+  isOpen,
+  selectedItem,
+  setSelectedItem,
+  setIsModalOpen,
+  isCreateModal = false,
+  item_id = '',
+  restaurant_id = '',
+}: ModalMenuItemPropsType) => {
+  const {
+    control,
+    handleSubmit,
+    reset,
+    formState: { isDirty, errors },
+  } = useForm({
+    defaultValues: {
+      name: selectedItem?.name || '',
+    },
+    resolver: zodResolver(menuListInfoSchema),
+  });
+
+  const [imageFile, setImageFile] = useState<File | null>(null);
+  const [imageUrl, setImageUrl] = useState<string | null>(null);
+  const [uploadError, setUploadError] = useState<string | null>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isConfirmModalOpen, setIsConfirmModalOpen] = useState<boolean>(false);
+
+  const { mutate: updateMenuItem } = useUpdateMenuItemApi({ item_id });
+  const { mutate: deleteMenuItem } = useDeleteMenuItemApi(item_id);
+  const { mutate: createMenuItem } = useCreateMenuItemApi();
+
+  const queryClient = useQueryClient();
+
+  const handleSave = (data: { name: string }) => {
+    const payload: UpdateMenuItemPayloadType = {
+      ...data,
+      file: imageFile,
+    };
+
+    updateMenuItem(payload, {
+      onSuccess: (res) => {
+        console.log({ res });
+        queryClient.invalidateQueries({ queryKey: ['getMenuList'] });
+        toast.success(res.message);
+        setIsEditing(false);
+        handleCloseModal();
+      },
+      onError: () => {
+        toast.error('Update restaurant failed!');
+      },
+    });
+  };
+
+  const handleCancel = () => {
+    setIsEditing(false);
+    reset({
+      name: selectedItem?.name || '',
+    });
+  };
+
+  const handleClickAction = () => {
+    setIsConfirmModalOpen(true);
+  };
+
+  const handleDeleteItem = () => {
+    deleteMenuItem(undefined, {
+      onSuccess: (data) => {
+        toast.success(data.message);
+        setIsModalOpen(false);
+        queryClient.invalidateQueries({ queryKey: ['getMenuList'] });
+      },
+      onError: () => {
+        toast.error('Delete menu item failed!');
+      },
+    });
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
```

Comment:
when closing the modal, all the states inside component will be destroyed. We don't need to reset state